pycbc/types/optparse.py

```diff
@@ -162,6 +162,36 @@ def __call__(self, parser, namespace, values, option_string=None):
                 raise ValueError(err_msg)
         setattr(namespace, self.dest, items)
 
+class MultiDetMultiColonOptionAction(MultiDetOptionAction):
+    """A special case of `MultiDetOptionAction` which allows one to use
+    arguments containing colons, such as `V1:FOOBAR:1`. The first colon is
+    assumed to be the separator between the detector and the argument.
+    All subsequent colons are kept as part of the argument. Unlike
+    `MultiDetOptionAction`, all arguments must be prefixed by the
+    corresponding detector.
+    """
+    def __call__(self, parser, namespace, values, option_string=None):
+        err_msg = ('Issue with option: {}\n'
+                   'Received value: {}\n').format(self.dest, ' '.join(values))
+        if getattr(namespace, self.dest, None) is None:
+            setattr(namespace, self.dest, {})
+        items = copy.copy(getattr(namespace, self.dest))
+        for value in values:
+            if ':' not in value:
+                err_msg += ("Each argument must contain at least one ':' "
+                            "character")
+                raise ValueError(err_msg)
+            detector, argument = value.split(':', 1)
+            if len(detector) != 2:
+                err_msg += 'The detector name must be 2 characters long'
```

Comment:
There's no need to enforce that the detector name be 2 characters long. I think Alex wanted to allow this to be arbitrary in general, but there's no need to enforce it here given the rest of this block ... If you're trying to catch the case that you want the code to fail if someone does mess this up maybe add a logging.warn here if detector name is longer than say 5 characters that this is unusual, and perhaps an error? (Could also do that in the other MultiIfo Option classes).